test/unit/power_manager_test.rb

```diff
@@ -42,4 +43,51 @@ class PowerManagerTest < ActiveSupport::TestCase
       assert_includes PowerManager::SUPPORTED_ACTIONS, action
     end
   end
+
+  test "should respond correctly to status when compute resource is started" do
+    vm_mock = mock('vm')
```

Comment:
lines 48 to 57 can go in a setup block, they're the same for both tests - just keep `vm_mock.expects(:state).at_least_once.returns('XXXX')` in the tests themselves